cmd/dotmesh-server/pkg/main/controller.go

```diff
@@ -783,9 +783,10 @@ func (s *InMemoryState) CreateFilesystem(
 		return nil, nil, err
 	}
 
-	// TODO: update our own knowledge of mastersCache with what we just
-	// updated, so that we don't have to wait for the echo from etcd (which
-	// causes race conditions)
+	//update mastersCache with what we know
```

Comment:
It might be wise to wrap this new section in `func() { ... }()` so the defer triggers as soon as we're done with `mastersCache`, to avoid holding that lock while we do the rest of the code in `CreateFilesystem`.